### PR TITLE
Desktop: Resolves #3535: Configure tinymce to handle the first table row as header

### LINKED ISSUE #3535: WYSIWYG - Table gets header row added after switching notes
## Environment

Joplin version: 1.0.227
Platform: Linux
OS specifics: Pop!_OS 20.04

## Steps to reproduce

1. Create a simple table, 2 rows, 2 columns
2. Enter some data in the cells
3. Switch to another note
4. Switch back to the note with the table
5. Verify a new (top) row has been added to the table

## Describe what you expected to happen
There should not be an additional row be added, the table should contain the top row as part of the creation process.

## Screencast
Added top row Joplin

### PR DESCRIPTION
**Pull request by marph91**
Resolves #3535

Consider the first table row as header in the richtext editor automatically. Switching isn't needed anymore. This is based on https://tinymce.github.io/tinymce-demos/tables/customizing-the-table-toolbar.html.

Open questions:

1. The new menu only contains the table size selection. As far as I see, most of the options are either covered by the table tooltip (insert, delete, properties) or incompatible with markdown (merge and split cells). Only the cut/paste options are missing. I could either try to add the cut/paste options to the tooltip or build the menu manually, as it was before. What would be the preferred way?
2. There are some `any` types. I left them for now, since the surrounding code has them, too. Let me know if I should try to fix them.

Screenshot:

Screenshot_20230511_155150

### CODE REVIEW

**Comment by marph91** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by laurent22** [comment]
Thank you, that looks very good. The menu we have at the moment is indeed needlessly complex.

Regarding `any`, please could you check if you can get the types from TinyMCE? If too complicated, don't worry, but in general for new code we try to set types everywhere

**Comment by marph91** [comment]
I added the missing functions to the table toolbar. Some were removed, because they were non-functional or duplicated. This is how it looks now:

Screenshot_20230516_182656

This allows for example easy copy pasting of rows/columns by only three clicks.

Types should be complete now, too.

File: packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx [hunk 1]
```diff
@@ -557,7 +558,15 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 				'bold', 'italic', 'joplinHighlight', 'joplinStrikethrough', 'formattingExtras', '|',
 				'link', 'joplinInlineCode', 'joplinCodeBlock', 'joplinAttach', '|',
 				'bullist', 'numlist', 'joplinChecklist', '|',
-				'h1', 'h2', 'h3', 'hr', 'blockquote', 'table', `joplinInsertDateTime${toolbarPluginButtons}`,
+				'h1', 'h2', 'h3', 'hr', 'blockquote', 'inserttable', `joplinInsertDateTime${toolbarPluginButtons}`,
+			];
+
+			// Available table toolbar buttons:
+			// https://www.tiny.cloud/docs/advanced/available-toolbar-buttons/#tableplugin
+			const table_toolbar = [
```
Comment (laurent22) [inline packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx]: tableToolbar (see coding_style.md)

**Review by laurent22 — COMMENTED** [review]
Just one minor issue and we can merge. Thanks for clearing the "any" types!

**Review by laurent22 — COMMENTED** [review]
Just one minor issue and we can merge. Thanks for clearing the "any" types!

File: packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx (hunk 1 above)
Comment (marph91) [reply packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx]: Should be fixed now. I guess the TinyMCE parameter name confused me.

**Comment by laurent22** [comment]
Thanks for implementing this @marph91!

**Pull request merged**